File: server/Models/Post.cs
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace server.Models
{
    public class Post
    {
        public int Id { get; set; }
        public string Text { get; set; }
        // public string Image { get; set; }
        public DateTime Timestamp { get; set; } = DateTime.Now;
        public int Likes { get; set; } = 0;
        public ICollection<PostComment> Comments { get; set; } = new List<PostComment>();
        public User User { get; set; }
        [JsonIgnore]
        public int UserId { get; set; }
        public bool LikedByUser { get; set; } = false;
    }
}

File: client/src/models/post.ts
import { PostComment } from "./postComment";
import { User } from "./user";

interface UserInfo {
    id: number,
    username: string
}

export class Post {
    constructor(
        private _id: number,
        private _text: string,
        private _timestamp: Date,
        private _comments: PostComment[],
        private _likesCount: number,
        private _user: UserInfo
    ){  }

    public get id()         : number        { return this._id }
    public get text()       : string        { return this._text }
    public get timestamp()  : Date          { return this._timestamp }
    public get comments()   : PostComment[] { return this._comments }
    public get likesCount() : number        { return this._likesCount}
    public get user()       : UserInfo      { return this._user }
}